[thisisaditya17}] iP

thisisaditya17 opened the pull request:
### KillJoy

KillJoy, your tech-savvy personal assistant, brings efficiency and a touch of sass to task management. It's:

- **command line interface**
- **simple to use**
- **SMART & FUN**

Just:

1. **Download it from here.**
2. **Run it.**
3. **Tell KillJoy what you need.**
4. **Sit back and let KillJoy organize your life with a playful banter. 😉**

And yes, it's **FREE!**

#### Features:

- **Task management**
- **Playful interaction**

For those who love coding, KillJoy is a fun way to practice your programming skills. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        KillJoy kj = new KillJoy();
        kj.start();
    }
}
```

**Let KillJoy take the load off your mind and put a smile on your face!**

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

thisisaditya17 committed:
KillJoy.java: update chatbot name and implement basic skeleton

- Rename the chatbot from 'Duke' to 'KillJoy'.
- Implement a skeletal version of the chatbot, including greeting and exit functionalities.

thisisaditya17 committed:
README.md: update with the new name

-Update README with the new assistant name and ASCII art

thisisaditya17 committed:
KillJoy.java: echo user inputs and exit on "bye"

- Implement functionality to echo commands entered by the user.
- Add logic to terminate the chatbot session when the user types the command 'bye'.

thisisaditya17 committed:
Killjoy.java: add text storage and retrieval feature

- Add the ability to store user's tasks
- Add feature to display stored tasks on user's request

thisisaditya17 committed:
add Task.java

thisisaditya17 committed:
KillJoy.java: implement mark and unmark task feature

thisisaditya17 committed:
README.md: add a fun introduction

thisisaditya17 committed:
Task.java: change toString

thisisaditya17 committed:
add new classes inherited from Task.java

thisisaditya17 committed:
KillJoy.java: change to implement todo, deadline, event for tasks

thisisaditya17 committed:
KillJoy.java: fix some bad logic

thisisaditya17 committed:
runtest.sh: change the location to run the test

thisisaditya17 committed:
add input and expected text to run test

thisisaditya17 committed:
KillJoy.java: add error handling for invalid inputs

thisisaditya17 committed:
change input and expected to accommodate changes

thisisaditya17 committed:
KillJoy.java: add delete task feature

thisisaditya17 committed:
change input and expected text to test delete task

thisisaditya17 committed:
README.md: update README.md with commands

thisisaditya17 committed:
add ProcessTask class for processing tasks of KillJoy

thisisaditya17 committed:
KillJoy.java: update KillJoy to use ProcessTask class for processing

thisisaditya17 committed:
create new class SaveAndLoad

thisisaditya17 committed:
include save and load tasks methods in KillJoy class

thisisaditya17 committed:
fix logical error

thisisaditya17 committed:
Merge branch 'branch-Level-7'

thisisaditya17 committed:
use Date Time libraries to store deadline dates in a format

thisisaditya17 committed:
Merge branch 'branch-Level-8'

thisisaditya17 committed:
change name of SaveAndLoad to Storage

thisisaditya17 committed:
create Parser class to handle parsing methods

thisisaditya17 committed:
create UserInferface class to handle all ui tasks

thisisaditya17 committed:
change KillJoy and ProcessTasks class to include new classes

thisisaditya17 committed:
change to handle exceptions and errors

thisisaditya17 committed:
Merge branch 'branch-A-MoreOOP'

thisisaditya17 committed:
add all classes in a package

thisisaditya17 committed:
Merge branch 'branch-A-Packages'

thisisaditya17 committed:
Merge branch 'add-gradle-support'

thisisaditya17 committed:
add Gradle to automate project builds

thisisaditya17 committed:
add junit tests to test Parser and Task class

thisisaditya17 committed:
add Java docs to all the java files

thisisaditya17 committed:
Merge branch 'branch-A-JavaDoc'

thisisaditya17 committed:
make changes in the code to adhere to Java coding standard

thisisaditya17 committed:
Merge branch 'A-CodingStandard'

# Conflicts:
#	src/main/java/killjoy/task/Todo.java

thisisaditya17 committed:
add findTask method to find tasks using a keyword

thisisaditya17 committed:
Merge branch 'branch-Level-9'

SwaminathanViswa commented on src/main/java/killjoy/main/KillJoy.java:
```diff
@@ -0,0 +1,134 @@
+    }
+
+    /**
+     * Adds a task to the task list.
```
There should be an empty line between the description and the parameter section. The same goes for the rest of the javadocs :)

SwaminathanViswa commented on src/main/java/killjoy/main/KillJoy.java:
```diff
@@ -0,0 +1,134 @@
+
+        saveAndLoad.loadTasks();
+
+        while(true) {
```
there should be a space after while 💭

SwaminathanViswa commented on src/main/java/killjoy/processing/ProcessTasks.java:
```diff
@@ -0,0 +1,200 @@
+
+        ui.printLine();
+        switch (taskType) {
+        case TODO: {
```
According to the coding standard website [here](https://se-education.org/guides/conventions/java/intermediate.html), there is no need for a pair of curly braces {} after each case. 😄

thisisaditya17 replied:
I asked prof about it. He said that if it is not specified you can do it according your own liking but it has to be consistent. The thing is without braces, the variables declared can be used by other cases as well. To prevent it, I enclosed them in {}.

SwaminathanViswa commented on src/main/java/killjoy/processing/Parser.java:
```diff
@@ -0,0 +1,139 @@
+     * @param input The user input.
+     * @return The to date of the task.
+     */
+    public static String getTo(String input) {
```
maybe getToString sounds better than getTo as getTo sounds like its going from one location to another? 🤔

sumomomomomo replied:
I agree, "getTo" can be ambiguous. However I would suggest renaming it to "getToDate".

SwaminathanViswa commented on src/main/java/killjoy/processing/Parser.java:
```diff
@@ -0,0 +1,139 @@
+     * @param input The user input.
+     * @return The from date of the task.
+     */
+    public static String getFrom(String input) {
```
Similar to getTo, getFrom could be renamed as getFromString or getFromDeadline

SwaminathanViswa commented on src/main/java/killjoy/processing/Parser.java:
```diff
@@ -0,0 +1,139 @@
+     * @param input The user input.
+     * @return The by date of the task.
+     */
+    public static String getBy(String input) {
```
Similar to getTo, getBy could be renamed as getByString or getByDeadline

Justin-Yeo replied:
i think the naming of the method could be something clearer to prevent ambiguity!

SwaminathanViswa reviewed (commented):
Overall it is very good and only a bit of changes to be made to ensure code quality! Well done on the code namings as majority were well named and easy to understand! 😃

thisisaditya17 committed:
.gitignore: add unwanted directories

thisisaditya17 committed:
clean up unwanted files from repository

thisisaditya17 committed:
create killjoy.jar file for the chatbot

sumomomomomo commented on src/main/java/killjoy/main/UserInterface.java:
```diff
@@ -0,0 +1,109 @@
+        System.out.println("    Yo Dawgg!! Added this task:");
+        System.out.println("    " + killJoy.getTask(killJoy.getTaskCount() - 1));
+        if (killJoy.getTaskCount() == 1) {
+            System.out.println("    Now you have " + killJoy.getTaskCount() + " task in the list.");
```
You could replace "killjoy.getTaskCount()" with "1".
```suggestion
            System.out.println("    Now you have 1 task in the list.");
```

thisisaditya17 replied:
May I know what would be the purpose of it?
I mean will this make a difference?😅😅

sumomomomomo commented on src/main/java/killjoy/processing/Parser.java:
```diff
@@ -0,0 +1,139 @@
+     * @param dateTime The date and time.
+     * @return The date and time.
+     */
+    public static LocalDateTime parseDateTime(String dateTime) {
```
What "parse" means could be made clearer. Parse here returns a LocalDateTime, but parse below in parseTaskInfo returns a String[]. Maybe consider something like "parseStringToLocalDateTime".
```suggestion
    public static LocalDateTime parseStringToLocalDateTime(String stringDateTime) {
```

sumomomomomo commented on src/main/java/killjoy/processing/ProcessTasks.java:
```diff
@@ -0,0 +1,200 @@
+     * @param taskInfo The task information.
+     */
+    public void createTasks(String taskInfo) {
+        String[] parts = Parser.parseTaskInfo(taskInfo);
```
I like that you named it as the plural form "parts".

sumomomomomo commented on src/main/java/killjoy/task/Deadline.java:
```diff
@@ -0,0 +1,32 @@
+ */
+public class Deadline extends Task {
+    private LocalDateTime by;
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd MMM yyyy HH:mm:ss");
```
I feel "FORMATTER" is ambiguous. Looking at your code, its only purpose is to format the date in toString(). Consider:
```suggestion
    private static final DateTimeFormatter TO_STRING_FORMATTER = DateTimeFormatter.ofPattern("dd MMM yyyy HH:mm:ss");
```
Another small nit but consider moving FORMATTER to Task so you don't redefine it in every subclass of Task.

sumomomomomo reviewed (commented):
Generally good. Suggested some changes to make naming less ambiguous.

Justin-Yeo commented on src/main/java/killjoy/main/UserInterface.java:
```diff
@@ -0,0 +1,109 @@
+    public UserInterface(KillJoy killJoy) {
+        this.killJoy = killJoy;
+    }
+
+    public String getMarkString() {
+        return MARK_STRING;
+    }
+
+    public String getUnmarkString() {
+        return UNMARK_STRING;
+    }
+
+    public String getDeleteString() {
+        return DELETE_STRING;
+    }
+
+    public String getExitString() {
+        return EXIT_STRING;
+    }
+
+    public String getWelcomeString() {
+        return WELCOME_STRING;
+    }
+
+    public String getLogoString() {
+        return LOGO_STRING;
+    }
```
i like the clear naming of the methods! 💯

Justin-Yeo commented on src/main/java/killjoy/main/KillJoy.java:
```diff
@@ -0,0 +1,134 @@
+package killjoy.main;
+
+import killjoy.task.Deadline;
+import killjoy.task.Event;
+import killjoy.task.Task;
+import killjoy.task.Todo;
+import killjoy.processing.ProcessTasks;
+import killjoy.processing.Storage;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Scanner;
```
if im not mistaken the ordering of imports are wrong, just a slight change! 👍

thisisaditya17 replied:
Can you specify what type of ordering should i follow?

Justin-Yeo commented on src/main/java/killjoy/task/Task.java:
```diff
@@ -0,0 +1,63 @@
+    /**
+     * Changes the status of the task.
+     */
+    public void changeStatus() {
+        if (this.isDone) {
+            this.isDone = false;
+        } else {
+            this.isDone = true;
+        }
+    }
```
i like the naming but perhaps could give greater details by naming it as changeTaskStatus instead?

Justin-Yeo reviewed (commented):
great work! just minor comments on the naming 💯

thisisaditya17 committed:
apply feedback to improve code quality

shadhankkk commented on src/main/java/killjoy/task/Task.java:
```diff
@@ -0,0 +1,66 @@
+        return this.description;
+    }
+
+    public boolean getTaskStatus() {
```
maybe you can change this boolean function name to something that follows the is, has, was, etc. format

e.g: isTaskActive() or something like that

shadhankkk commented on src/main/java/killjoy/task/Event.java:
```diff
@@ -0,0 +1,30 @@
+    public String getTaskInfo() {
+        return "EVENT|" + String.valueOf(isDone ? 1 : 0) + "|" + this.description + "|" +
+                String.valueOf(this.from).replace("T", " ") + "|"
+                + String.valueOf(this.to).replace("T", " ") + "\n";
```
I like your use of line breaks here

shadhankkk reviewed (commented):
Great code overall!

thisisaditya17 committed:
add checkstyle config files

thisisaditya17 committed:
fix all the styling errors pointed by checkstyle

thisisaditya17 committed:
add new classes to run JavaFX

thisisaditya17 committed:
add resources for JavaFx

thisisaditya17 committed:
KillJoy.java: change to work as an interface between Main and other classes

thisisaditya17 committed:
change to return string and not print the strings

thisisaditya17 committed:
add new resource images

thisisaditya17 committed:
update font in css files and modify welcome message content

thisisaditya17 committed:
add DialogBoxImg to handle image in dialog boxes

thisisaditya17 committed:
change code to accommodate DialogBoxImg

thisisaditya17 committed:
change font style, colors and few resources

thisisaditya17 committed:
fix few logical errors

thisisaditya17 committed:
fix all the styling errors

thisisaditya17 committed:
Merge branch 'branch-Level-10'

thisisaditya17 committed:
generate new JAR file with Gradle

thisisaditya17 committed:
Use assert feature to document important assumptions

thisisaditya17 committed:
Merge pull request #1 from thisisaditya17/branch-A-Assertions

Use assert feature

thisisaditya17 committed:
Main.java: Refactor code to adhere to code quality

Refactored the start() method into multiple smaller methods (setupStage(), loadFXML(), initializeScene(), configureController(), showStage()) for better structure and easier debugging.

Replaced magic numbers for stage dimensions with named constants (MIN_STAGE_HEIGHT and MIN_STAGE_WIDTH) for better maintainability.

thisisaditya17 committed:
ProcessTask.java: Refactor code to adhere to code quality

Refactored large, switch-based methods into specific task-handling methods (e.g., handleDeadlineTask) to make each case more modular and easier to understand.

Introduced proper error handling for invalid input scenarios without returning error messages from void methods.

Improved separation of concerns by delegating task-creation logic to dedicated methods, removing complexity from processUserInput().

thisisaditya17 committed:
Storage.java: Refactor code to adhere to code quality

Simplified loadTasks() by abstracting file creation logic into separate methods, making the code easier to follow and reducing complexity.

Improved error handling in saveTasks() by consolidating the file-writing logic into a single, reusable method, reducing duplication and improving reliability.

thisisaditya17 committed:
change if-else statements to switch statement for better code quality

thisisaditya17 committed:
refactor code to adhere to code quality

thisisaditya17 committed:
Merge branch 'master' into branch-A-CodeQuality

thisisaditya17 committed:
insert removed assert statements back

thisisaditya17 committed:
Merge pull request #2 from thisisaditya17/branch-A-CodeQuality

Merge code quality branch with Master

thisisaditya17 committed:
add archive directory

thisisaditya17 committed:
fix few bugs

thisisaditya17 committed:
add archive and load archive commands

thisisaditya17 committed:
add new load and archive messages

thisisaditya17 committed:
add archive and load archive methods

thisisaditya17 committed:
fix a few bugs due to changes

thisisaditya17 committed:
Merge branch 'Branch-BCD-Extension'

thisisaditya17 committed:
update README.md to accommodate new changes

thisisaditya17 committed:
enable assertions in build.gradle

thisisaditya17 committed:
insert assert statement for archive command

thisisaditya17 committed:
insert screenshot of ui

thisisaditya17 committed:
change font

thisisaditya17 committed:
update README.md

thisisaditya17 committed:
insert comment

thisisaditya17 committed:
insert stage title

thisisaditya17 committed:
add new invalid command format message for DateTime tasks

thisisaditya17 committed:
fix a few bugs and update invalid command format message

thisisaditya17 committed:
Add comprehensive User Guide to README.md

Update README.md to include a detailed User Guide for KillJoy.
- Add usage instructions for all key features including:
  - Adding TODOs, Deadlines, and Events.
  - Listing tasks.
  - Marking and unmarking tasks.
  - Deleting tasks.
  - Archiving completed tasks with unique file naming and folder creation.
  - Loading tasks from an archived file.
  - Finding tasks by keyword.
  - Exiting the KillJoy application.
- Include command examples and expected outcomes for better clarity.

thisisaditya17 committed:
remove Java Jar files from repo

thisisaditya17 committed:
Rename ui.png to Ui.png

thisisaditya17 committed:
update docs/README.md to include how to test killjoy.jar file

thisisaditya17 committed:
Update docs/README.md to accommodate Ui.png

thisisaditya17 committed:
update dependencies to support windows and linux

thisisaditya17 committed:
split changeStatus method into two different methods

thisisaditya17 committed:
update Invalid command message

thisisaditya17 committed:
fix all the methods that referenced changeStatus method

thisisaditya17 committed:
fix String bug for Event tasks

thisisaditya17 committed:
fix invalid command format message
